trace/trace.go

```diff
@@ -472,3 +580,39 @@ func (s *Span) createChildSpan(ctx context.Context, async bool) (context.Context
 	ctx = PutSpanInContext(ctx, newSpan)
 	return ctx, newSpan
 }
+
+type randomIdentifier struct {
+	randomBytes   func([]byte) (int, error)
+	incrementLock sync.Mutex
+	nonRandomId   uint64
+}
+
+// newRandomIdentifier returns a new randomIdentifier with the default pseudo-random
+// generator.
+func newRandomIdentifier() *randomIdentifier {
```

Comment:
naming nit - the name of this function makes me think it's going to return an ID itself.  but it doesn't, it returns an ID generator.

Reply:
Renamed to `newIDGenerator` and renamed the struct to `idGenerator`.